### PR TITLE
Expand documentation about compile command inference for header files

### PR DESCRIPTION
**Pull request by HighCommander4**

### CODE REVIEW

**Comment by HighCommander4** [comment]
@kadircet I did see you comment on the commit:

> i don't think it is worth describing some intermediate state in these docs

but I'm unclear which part of the patch it applies to.

I think adding mention of the CompDB workaround is important. It has come up over and over again in support channels (recall, I [made a list](https://github.com/clangd/clangd/issues/123#issuecomment-883162461) :-), and there have been additional instances since then), and this workaround solves the problem for many users.

File: design/compile-commands.md [hunk 1]
```diff
@@ -100,6 +100,9 @@ compile commands for them.
 If no command is available for a header, but a file has been opened that
 transitively includes it, then that file's command will be used. To enable this
 [TUScheduler] keeps a `HeaderIncluderCache` to look up the relevant file.
+[This issue](https://github.com/clangd/clangd/issues/123) tracks a further
```
Comment (kadircet) [inline design/compile-commands.md:103]: I was talking about the delta here. We should update the docs once we have some inference through Index, but I don't think it is useful to talk about this intermediate state here.

**Review by kadircet — CHANGES_REQUESTED** [review]
> I think adding mention of the CompDB workaround is important. This issue has come up over and over again in support channels (recall, I made a list :-), and there have been additional instances since then), and the workaround solves the problem for many users.

Right, I wasn't objecting to the FAQ entry. It was for the change in the design document.

File: design/compile-commands.md (hunk 1 above)
Comment (HighCommander4) [reply design/compile-commands.md]: Thanks, removed this change.

File: faq.md [hunk 2]
```diff
@@ -123,4 +123,17 @@ CompilationDatabase: <path>
...
+## What can I do if clangd chooses the wrong source file to infer commands for a header?
+
+Clangd currently uses
+[heuristics](https://clangd.llvm.org/design/compile-commands#commands-for-header-files)
+to choose a source file whose compile command to use when opening a header file.
+
+Sometimes these heuristics choose the wrong file. While 
+[improvements](https://github.com/clangd/clangd/issues/123) to the heuristics
+are planned, a workaround in the meantime is to use a tool like
+[CompDB](https://github.com/Sarcasm/compdb#generate-a-compilation-database-with-header-files)
```
Comment (kadircet) [inline faq.md:137]: i'd probably put the remedy at top. (e.g. You can use CompDB to have flags for headers.) Then explain clangd uses heuristics and they are constantly being improved (with the link to thread).

**Review by kadircet — APPROVED** [review]
thanks, LGTM!

File: faq.md (hunk 2 above)
Comment (HighCommander4) [reply faq.md:131]: Thanks, revised as suggested.

**Comment by HighCommander4** [comment]
@kadircet I don't have write permissions in this repository, so please go ahead and merge if you're happy with the update!

**Pull request merged**